src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -368,6 +368,38 @@ def get_package_size(self, output):
     # endregion
 
     # region auto OS updates
+    def get_current_auto_os_patch_state(self):
+        """ Gets the current auto OS update patch state on the machine """
+        self.composite_logger.log("Fetching the current automatic OS patch state on the machine...")
+        self.__get_current_auto_os_updates_setting_on_machine()
+        if self.unattended_upgrade_value != 0 or self.unattended_upgrade_value != 1:
+            return Constants.PATCH_STATE_UNKNOWN
+        elif self.unattended_upgrade_value:
+            return Constants.PATCH_STATE_ENABLED
+        else:
+            return Constants.PATCH_STATE_DISABLED
```

Comment:
The logic may be a bit simpler if you re-order the checks:

if self.unattend_upgrade_value == 0
    return Constants.PATCH_STATE_DISABLED
elfi self.unattend_upgrade_value == 1
    return Constants.PATCH_STATE_ENABLED
else
    return Constants.PATCH_STATE_UNKNOWN

Reply:
Resolved